build-and-stage.ps1

```diff
@@ -1,205 +1,26 @@
-    [string]$PackageDisplayName = "",                        # The human-readable name of the package (ex. "openssl (static)").  If not specified, it will default to "$Package-$LinkType"
+    [string]$VcpkgHash = "",                                 # The hash of vcpkg to checkout (if applicable)
     [switch]$ShowDebug = $false                              # Show additional debugging information
 )
 
-$vcpkgRepo = "https://github.com/TechSmith/vcpkg.git"
-. "$PSScriptRoot\util.ps1"
+$global:showDebug = $ShowDebug
+Import-Module "$PSScriptRoot/ps-modules/Build" -Force -DisableNameChecking
 
-Write-Host "************************************************************"
-Write-Host "Starting vcpkg install for: $PackageAndFeatures"
-Write-Host "************************************************************"
-$allParams = @{
-    PackageAndFeatures = $PackageAndFeatures
-    BuildType = $BuildType
-    LinkType = $LinkType
-    StagedArtifactsPath = $StagedArtifactsPath
-    ReleaseTagBaseName = $ReleaseTagBaseName
-    PackageName = $PackageDisplayName
-    ShowDebug = $ShowDebug
-}
-Write-Host "Parameters:"
-foreach ($paramName in $allParams.Keys) {
-    $paramValue = $allParams[$paramName]
-    Write-Host "- $paramName`: $paramValue"
-}
+Write-Banner -Level 2 -Title "Starting vcpkg install for: $PackageAndFeatures"
+Write-Message "Params:"
+Write-Message (Get-PSObjectAsFormattedList -Object $PSBoundParameters)
 
-Write-Host ""
-Write-Host "============================================================"
-Write-Host "Initializing"
-Write-Host "============================================================"
-$packageNameOnly = Get-PackageNameOnly -PackageAndFeatures $PackageAndFeatures
-if( $ReleaseTagBaseName -eq "") {
-   $ReleaseTagBaseName = "$packageNameOnly-$LinkType"
-}
-if( $PackageDisplayName -eq "") {
-   $PackageDisplayName = "$packageNameOnly-$LinkType"
-}
-$IsOnWindowsOS = $false # Built-in IsWindows variable not working on Windows agent
-$IsOnMacOS = $false
-Write-Host "Checking OS..."
-if ($env:OS -like '*win*') {
-    $IsOnWindowsOS = $true
-    Write-Host "> Running on: Windows"
-} else {
-    $IsOnMacOS = $true
-    Write-Host "> Running on: Mac"
-}
+$vars = Initialize-Variables -packageAndFeatures $PackageAndFeatures -linkType $LinkType -packageName $PackageName -buildType $BuildType -stagedArtifactsPath $StagedArtifactsPath -vcpkgHash $VcpkgHash
+Setup-VcPkg -repo $vars.vcpkgRepo -repoHash $vars.vcpkgRepoHash -installDir $vars.vcpkgInstallDir -cacheDir $vars.vcpkgCacheDir -bootstrapScript $vars.vcpkgBootstrapScript
+Run-PreBuildScriptIfExists -script $vars.prebuildScript
+Install-Package -vcpkgExe $vars.vcpkgExe -package $PackageAndFeatures -triplets $vars.triplets
+Create-FinalArtifacts -x64Dir $vars.macX64Dir -arm64Dir $vars.macArm64Dir -preStagePath $vars.preStagePath
```

Comment:
Since we are already passing mac-specific directories here, maybe we just do the MacOS check here and just call `Create-FinalMacArtifacts` directly.

If I were walking through this script from a Windows-perspective, I would be confused by mac dirs are being passed in here.

Reply:
Yeah, kind of related to https://github.com/TechSmith/ThirdParty-Packages-vcpkg/pull/1#discussion_r1356879193.  I think I'm going to work on just narrowing the interfaces on all steps in this way.

It was kind of a short-sighted one-level refactoring that I didn't take another pass at and probably should have.